### ISSUE TITLE
Xamarin.iOS 17.x release date or any beta version available to support iOS 17 Beta

### ISSUE THREAD

**Issue by mollea3** [xamarin/xamarin-macios#18409]
Hi Team,

We are working on the xamarin application which requires the latest toolchain along with the latest iOS Release, So the issue is the xamarin.iOS is not supported to create the IPA file for the iOS 17 Beta

Please support us by knowing the Release date of the Xamarin.iOS or a Beta Version to Support the Xcode 15 Beta.

[https://docs.microsoft.com/en-us/xamarin/ios/release-notes/] is not updating with the latest version updates anymore.

Please let us know when will be the Tentative date for the Xamarin.iOS Support for Xamarin 17 Support.

**Comment by TobiasJoergensen** [comment]
Hi Mollea3,
I understand there won't be a Xamairn.iOS 17 per EOL documentation.

[https://dotnet.microsoft.com/en-us/platform/support/policy/xamarin#microsoft-support](url)

**Comment by JohnHDev** [comment]
That policy specifically states that Xcode 14 is the last version Xamarin will support. This is a big issue as .NET MAUI is not ready for us to migrate to yet, with bug fixes going into .NET 8 before even being considered for back porting to .NET 7.

**Comment by RobTF** [comment]
Yes, I agree the stance on iOS 17 should be changed - Xamarin needs support to buy time for MAUI to get properly stabilised.

Maybe a compromise might be that new iOS 17 specific APIs aren't surfaced, but the toolchain etc. should be supported.

**Comment by pasha-o** [comment]
I upvote this as well- We have projects that need to be maintained until Maui is stable @dalexsoto

**Comment by RobSchoenaker** [comment]
It shouldn't be too hard to even comment on this issue with a simple 'we are looking into this', right?

**Comment by dalexsoto** [comment]
Hello Everyone!

Sorry for the late response but I wanted to have all the information on hand before posting here. The plan stands, Xamarin.iOS/Mac won't have support for Xcode 15 as per our [Xamarin End of Support Policy](https://dotnet.microsoft.com/en-us/platform/support/policy/xamarin#microsoft-support).

> Xamarin support will end on May 1, 2024 for all Xamarin SDKs. Android 13 and Xcode 14 SDKs (iOS and iPadOS 16, macOS 13) will be the final versions Xamarin will target.

We will continue to service security fixes and new Xcode 14 Service releases if shipped. The way moving forward is migrating your apps to use .NET.

If you need anything else regarding this topic feel free to reach out to our Project Manager David Ortinau (david.ortinau at microsoft.com).

**Comment by JohnHDev** [comment]
Hi @dalexsoto, we can't migrate to .NET as .NET Maui isn't production ready with bugs being fixed in .NET 8 (which is in beta) rather than .NET 7.
The Microsoft approach to .NET Maui and the migration from Xamarin Forms is fundamentally flawed. Do you realise how difficult you are making the migration for the thousands of developers using Xamarin Forms? If we are forced to upgrade to .NET Maui to support iOS 17 (Xcode 15) before it is Maui is stable, then why don't we move to Flutter? The amount of work we have to do to work around layout issues and regressions could be offset by the new development work in rewriting in a better, supported product.

Ive worked with with Xamarin since it was Monotouch, yes I used to pay for this product. I have helped many, many developers over the years (mostly on the old forum which was shut down, didn't get the thank you badge that was promised, thankyouverymuch @jamesmontemagno), I have supported Xamarin with confidence that you'll get it right in the end. I now feel like Ive been stabbed in the back after supporting you guys for so long.

I now run an app development team consisting of 9 Xamarin developers + another 12 app developers using native and other technologies for a multi national company. I am **THIS CLOSE** to suggesting we migrate to one of those other technologies.

@davidortinau you didn't reply to my email from a few weeks back, is there any chance this decision will be reversed. All Xamarin developers need to have migrated to .NET Maui by 1st May 2024 and be stable by that date, as we won't be able to roll back or continue using Xamarin as Apple will stop signing Xcode 14 builds. We are 10 months out from a massive, massive problem, only MS can alleviate the pain.

**Comment by RobTF** [comment]
Hi,

Firstly, @dalexsoto thankyou very much for making it clear the intentions of the team, it is much appreciated.

From my perspective, I understand the position of the Xamarin/MAUI team but am disappointed with the lack of acknowledgement as to why customers are struggling to adopt MAUI/.NET over Xamarin in the first place. Simply stating that customers must move to .NET without facing the issues as to why they haven't head on leaves me wondering whether the view within Microsoft is either that the issues simply don't exist, aren't as serious as customers are stating, or that it's perhaps an awkward minority of "pro Xamarin" users complaining and they will get with the program in time.

My view is this; I think MAUI is fantastic and like Xamarin before it I feel as a framework/SDK/whatever it offers something truly unique to the market. In exasperation, members of the community sometimes write about migrating to Flutter or React Native however these are not like for like replacements for MAUI or Xamarin. What the team at Microsoft have done is allow development teams to short circuit the various wobbly dev tools supplied with the various mobile platforms and get existing .NET dev teams quickly up to speed writing mobile apps, and the value that adds has been tremendous to us, and I'm sure countless other teams as well. My team have released a number of commercial applications based on Xamarin which are used by thousands of users. One of those has been mostly successfully ported to MAUI.

That said, my observation over the last few years has been that the development process for Xamarin felt rather odd; as myself and my team watched new "Whizzbang" features such as new items controls, the shell API and other bits come along we welcomed them of course, but found it really odd that long-standing showstopping bugs simply went ignored. It almost felt as though the development process for Xamarin was guided by what the Xamarin developers found interesting to do, as opposed to what needed to be done.

When MAUI was announced my team and I really felt that this was a sign that Microsoft was going to put mobile development into top gear and start treating it as a first class part of .NET (such as ASP.NET) and the problems that plagued us with Xamarin would be dealt with. When MAUI was delayed for those months after the .NET 6 release we didn't care; our view was that we'd rather wait for things to get done right. The positive message we got at Build this year seemed to back this up.

With MAUI now being around 18 months old, it seems that it has had time to "bed in" and unfortunately the exact same problems that affected the Xamarin development process seem to have been inherited into .NET MAUI. Maybe this is unfair and internally things are quite different now, but from the outside it looks very similar.

So what's the issue? Why are community members complaining?

Well, I cannot speak for others, and really all I can write here is another rendition of the same stuff myself and others have been writing in countless other threads. To make a point, below is a small selection of what I would consider to be _basic_ functions that don't work in MAUI, and do work in Xamarin.

- There are a number of major UI issues which cause fundamental problems when developing any app which goes beyond trivial "hello world" examples.
  - Android threading issues when loading images [(Bug #14128)](https://github.com/dotnet/maui/issues/14128) which makes basic image rendering unstable and prone to failing or crashing the app. **This is not an issue in Xamarin**.
  - Android images disappearing in basic apps [(Bug #14471)](https://github.com/dotnet/maui/issues/14471) which makes basic image rendering glitchy and unusable. **This is not an issue in Xamarin**.
  - Basic list UI via CollectionView doesn't work on iOS. [(Bug #15164)](https://github.com/dotnet/maui/issues/15164). **This is not an issue in Xamarin**.
  - Basic geometry functions, such as rounded corners do not work correctly [(Bug #14927)](https://github.com/dotnet/maui/issues/14927). **This is not an issue in Xamarin**
  - Basic layout problems in iOS [(Bug #14926)](https://github.com/dotnet/maui/issues/14926). **This is not an issue in Xamarin**
- The development experience feels unfinished and feels dangerously flaky, for example;
  - The release process has no UI and requires developers to drop into the command line. **Xamarin had a UI and felt more polished**.
  - iOS release builds simply fail unexpectedly for non trivial apps without the developer putting a special previously unknown directive into the `.csproj` file. [(Bug #12519)](https://github.com/dotnet/maui/issues/12519). **This is not an issue in Xamarin**
  - The Mac connection pairing to PC takes 5-7x longer than Xamarin and keeps dropping out.
- Notifications simply don't work properly on Android [(Bug #14924)](https://github.com/dotnet/maui/issues/14924). **This is not an issue in Xamarin**
-  The "safe area" in iOS is simply broken [(Bug #2657)](https://github.com/dotnet/maui/issues/2657). **This is not an issue in Xamarin**
- The support story is really, really odd
   - .NET 6 is LTS but gets no MAUI updates.
   - .NET 7 is STS but gets only partial updates, missing some essential fixes.
   - .NET 8 preview is [getting bug fixes for major issues](https://github.com/dotnet/maui/releases), but is a preview. 

So when we're asked to update our Xamarin apps to .NET, my first question is _"which version?"_. It seems .NET 6 is no longer receiving any meaningful updates, from the evidence in the release notes it stands to reason that .NET 7 is playing second fiddle to .NET 8 preview, so that makes both of the officially supported versions of .NET seem like a bit of a non starter, right? Straight to .NET 8 preview then? Will that be fully supported when work on NET 9 preview starts? It's all very unclear.

So even with this small sample of issues we're in a position where images don't render properly, basic layouts are broken, release build doesn't work without resorting to undocumented tricks and developers are confused how much support they'll get. These aren't new feature requests or odd esoteric edge cases, **this is bread and butter stuff**.

My team and I have been working with Xamarin for production apps since 2017 and are fairly switched on with the ecosystem. Despite this we've managed to port one of our simpler apps across to MAUI using .NET 7, but our other apps are more complicated and we're hitting more and more of these MAUI specific issues that didn't exist in Xamarin we're spending hours trying to get things working and stable, will we be ready before Apple stop signing updates for our Xamarin app due to Xamarin not supporting XCode 15? I really can't be sure at this stage and it's very worrying.

The bottom line; Please understand: It is not that developers don't _want_ to move their development to MAUI. They simply _can't_ in a lot of cases.

If there is anything I can do to help, please let me know. If the MAUI team are honestly unaware of these problems or confused as to what they are, I'm happy to take the time to sit and chat with members of the team, show them our app and explain the problems we're facing to start a conversation and help the team understand at least some of the current customer pain points.

**Comment by RobSchoenaker** [comment]
I completely agree with @RobTF a @JohnHDev . When Apple pulls the plug on XCode 14, we are kind of left in the dark. For Xamarin, we do not need new features, we need to be able to publish new and updated apps to Apple and Google.
Preferrably with MAUI, but that is not stable yet - and even considering the amount of (extra) work to do to port Xamarin Forms apps to MAUI... I really want to migrate, but cannot at the moment due to things not working.

TL;DR: New iOS 17 features are not needed. XCode 15 support is kinda mandatory until MAUI is stable.

**Comment by giuseppenovielli** [comment]
Guys only a simple idea, if xcode 15 will not supported with MonoTouch runtime, but with .NET Runtime, could be possible migrate Xamarin.Forms code to .NET Runtime?
So you continue using Renderer architecture and all Xamarin.Forms stable architecture, but with .NET iOS, .NET Android?

**Comment by giuseppenovielli** [comment]
@RobTF no, sorry but there are a misunderstood.
I'm not an a developer of Xamarin/.NET MAUI Team! 
I am a developer that use Xamarin.Forms/.NET MAUI Tecnology.

My idea became not good, because, as you mention, also third library must migrate to .NET Runtime, over Xamarin.Forms.
So i think is too expensive.

**Comment by RobTF** [comment]
@giuseppenovielli Ah, no worries! :)

**Comment by pasha-o** [comment]
I also echo my worries about stopping Xamarin support too early. We do run enterprise software applications with many customers on Xamarin forms.  we have built applications that simply can't be migrated to Maui for the reasons already mentioned in this thread.
My request for Maui/Xamarin team @dalexsoto  and  @davidortinau is to have a direct word with the community over Microsoft Teams/ Zoom to hear our concerns directly as to why we can't move to Maui yet and to extend support for Xamarin.

Can you please acknowledge if this is possible?

**Comment by phillippschmedt** [comment]
I want to address our worries too. MAUI and the ecosystem does not seem to be stable enough to migrate our xamarin app as this point. We run into multiple issues with maui and also with 3rd party libraries that haven't caught up yet. Supporting at least the xcode 15 toolchain would be a big relief.

**Comment by RobTF** [comment]
I think some acknowledgement from the project team (@davidortinau ?) is really required here, even if just to acknowledge the issue of MAUI migration and customer pain points.

One of the most worrying parts of all this is the silence and seeming ignorance of the problem from the Microsoft side. Simply giving customers the marketing line of "Just update to MAUI" without acknowledging any of the comments in this thread clearly isn't enough. To maintain developer trust there needs to be assurances that there is a workable way forward here. This is a production-affecting problem for many customers.

Imagine if Azure was going to block applications running on ASP.NET 7 as they are deemed "legacy", and ASP.NET 8 was a complete rewrite with a bunch of new bugs due to it not being battle hardened and customers were struggling to update due to said bugs? Would this theoretical scenario fly? Is what's happening here any different?

**Comment by pasha-o** [comment]
Does anyone know when Apple will release the iOS 17 and when they no longer allow builds of Xcode 14 on the app store?

**Comment by phillippschmedt** [comment]
> Does anyone know when Apple will release the iOS 17 and when they no longer allow builds of Xcode 14 on the app store?

Usually the new major versions release in September. Regarding the app store guidelines the last time I saw something happening like this was around 6 months after the major release. So my guess would be around March 2024.

**Comment by JohnHDev** [comment]
> Does anyone know when Apple will release the iOS 17 and when they no longer allow builds of Xcode 14 on the app store?

iOS 17 should go gold in September as Phillip mentioned. Apple will stop signing Xcode 14 builds 1st May 2024 (according to David Ortinau).

Apple won't remove existing Xcode 14 apps from the store, you just won't be able to update them if you are still on Xamarin after that date, unless MS updates Xamarin to support Xcode 15 (or at least 'works with' Xcode 15).

**Comment by jowenece** [comment]
I just want to echo the sentiment of others on this thread and share the seriousness of the situation. I'd love to update our Xamarin apps to MAUI.  We're TRYING to update.  MAUI. IS. NOT. READY. for real apps.  What are we supposed to do as developers?  I think the alternative is clear, and if we leave we won't be back.

**Comment by jtorvald** [comment]
I've been using and promoting Xamarin since 2012 and completely agree with @RobTF and @JohnHDev

**Comment by giuseppenovielli** [comment]
I think the philosophy behind end Xamarin support, is that .NET 8 will be a stable version to upgrade to .NET MAUI.
So November 2023 you can migrate safe to .NET MAUI, with .NET 8.

I believe and hope!

But i think that .NET MAUI must integrate a release timing, also every month, but mostly according to issues severity.
For example if after a week of November 2023, there are 2 regression very important, we can't wait one month for these fixes and so at November 2023, we downgrade to .NET MAUI 7 and wait October 2023.

October 2023 there are other regression... so this could become a potential loop!

**Comment by kevinmiles** [comment]
This is sadly starting to sound a bit like the "silent treatment" in threads from the Silverlight disaster.
I guess some people never learn.

**Comment by RobSchoenaker** [comment]
FWIW and to who runs into this...., To get Xamarin iOS working on iOS 17 requires some mtouch arguments. The default compile will include NewsstandKit, which is deprecated. The following arguments need to be added:

`--weak-framework=NewsstandKit.framework/NewsstandKit`

Hope this helps some devs to at least continue for iOS for now...

**Comment by jtorvald** [comment]
@RobSchoenaker Dank! It seems that the issue is closed by @dalexsoto because of end of support policy, his reply quoted below:

> Hello Everyone!
> 
> Sorry for the late response but I wanted to have all the information on hand before posting here. The plan stands, Xamarin.iOS/Mac won't have support for Xcode 15 as per our [Xamarin End of Support Policy](https://dotnet.microsoft.com/en-us/platform/support/policy/xamarin#microsoft-support).
> 
> > Xamarin support will end on May 1, 2024 for all Xamarin SDKs. Android 13 and Xcode 14 SDKs (iOS and iPadOS 16, macOS 13) will be the final versions Xamarin will target.
> 
> We will continue to service security fixes and new Xcode 14 Service releases if shipped. The way moving forward is migrating your apps to use .NET.
> 
> If you need anything else regarding this topic feel free to reach out to our Project Manager David Ortinau (david.ortinau at microsoft.com).

In the mean time I'm trying to convert my apps to Maui, which at first sight does not seem to be a smooth process. It feels to me that it's best to start a Maui project and copy the code/screens over instead of upgrading a project.
If this mtouch argument helps us to build under iOS17 we can at least postpone the problems and win more time to upgrade.

**Comment by RobTF** [comment]
> It feels to me that it's best to start a Maui project and copy the code/screens over instead of upgrading a project.

@jtorvald that's exactly what we are doing. After reviewing the possibilities we determined this was the best approach, as you get the opportunity to move things over to the "MAUI" way of doing stuff and avoid any of the cruft/issues of Xamarin backwards compatability. We also took advantage of the MVVM community toolkit etc. at the same time vs. writing our own `BindableBase` class etc.

It's been said a number of times that MAUI has its stability and functionality issues, it would seem somewhat risky to further add a Xamarin compatibility layer to the problem if it can be avoided at all.

By compatibility layer I mean things like legacy components (`StackLayout` vs the new `Vertical/HorizontalStackLayout`, `Frame` vs `Border` as well as custom renderer/effect support amongst others.

**Comment by jtorvald** [comment]
@RobSchoenaker how did you manage to test your build?

If I download Xcode 15 and 
```
$ sudo xcode-select --switch ~/Downloads/Xcode-beta.app
$ xcode-select --version                              
xcode-select version 2397.
```
Then when I build:


```
_CompileToNative:
  /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/bin/mtouch @/Users/developer/Projects/Apps/test-app-xamarin/Test/obj/iPhone/AppStore/response-file.rsp --linkskip=Twilio.Video.iOS --linkskip=GSSDKScanFlow --linkskip=GSSDKCore --linkskip=GeniusScanSDK.ScanFlow --weak-framework=NewsstandKit.framework/NewsstandKit "--gcc_flags=-ObjC -lz -lsqlite3" 
  Xamarin.iOS 16.4.0.6 (xcode14.3: 97731c92c) using framework: /Users/developer/Downloads/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk
  Compilation failed with code 1, command:
  /Users/developer/Downloads/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -Wno-receiver-forward-class -Wno-objc-missing-super-calls -Wno-unguarded-availability-new -stdlib=libc++ -gdwarf-2 -std=c++14 -I/Library/Frameworks/Xamarin.iOS.framework/Versions/16.4.0.6/SDKs/MonoTouch.iphoneos.sdk/include -isysroot /Users/developer/Downloads/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.0.sdk -Qunused-arguments -miphoneos-version-min=12.0 -arch arm64 -c -o /Users/developer/Projects/Apps/test-app-xamarin/Test/obj/iPhone/AppStore/mtouch-cache/arm64/registrar.o -x objective-c++ /Users/developer/Projects/Apps/test-app-xamarin/Test/obj/iPhone/AppStore/mtouch-cache/registrar.m
  In file included from /Users/developer/Projects/Apps/test-app-xamarin/Test/obj/iPhone/AppStore/mtouch-cache/registrar.m:2:
  /Users/developer/Projects/Apps/test-app-xamarin/Test/obj/iPhone/AppStore/mtouch-cache/registrar.h:37:9: fatal error: 'NewsstandKit/NewsstandKit.h' file not found
  #import <NewsstandKit/NewsstandKit.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```
MTOUCH : error MT4109: Failed to compile the generated registrar code.
```

**Comment by RobSchoenaker** [comment]
> If I download Xcode 15 and

Don't! Use XCODE 14. As mentioned before, Xamarin will not be compatible with XCode 15 - that's what this whole threa dis about. I use XCode 14 and added this mtouch argument to get the compiled app to work with iOS 16 and iOS 17 preview

**Comment by JohnHDev** [comment]
Apple will stop signing Xcode 14 builds in May next year, either Xamarin needs to update to support building in Xcode 15 or we all have to migrate to .NET Maui before that date.

**Comment by jtorvald** [comment]
@JohnHDev yeah, that is what I feared. Need to see how I can convert a 11 year old Xamarin iOS/Android app with some Forms pages to Maui/.NET. Probably easiest to rebuild it in Maui

**Comment by code723** [comment]
For me it only works in debug mode. As soon as I switch to Release in Visual Studio and archive the app, it crashes again on iOS17.
Anyone have any ideas as to what could be causing this?

**Comment by jtorvald** [comment]
@code723  did you add the extra monotouch argument only in debug mode and not in release mode? 
`--weak-framework=NewsstandKit.framework/NewsstandKit`

**Comment by code723** [comment]
also in release mode

**Comment by zboldyga** [comment]
Any new developments on this? Another concerned party here, we have a critical app on Xamarin and looming Android and iOS version bumps may lead to us being unable to support thousands of healthcare providers.

**Comment by giuseppenovielli** [comment]
@zboldyga 
https://github.com/dotnet/maui/discussions/15540

**Comment by mhamsa** [comment]
We've been waiting for months and months and months for a workable "Upgrade Assistant". At this point there is no upgrade path for our complex Xamarin Forms applications. Our only options is to basically start over again with MAUI and salvage as much code as possible. Years of investments in Xamarin Forms development. Gone. Simply heartbreaking.

**Comment by pasha-o** [comment]
hi @mhamsa sorry for the pain and the agony - we are in the same boat. what areas did you find most struggling within your app tp migrate?

**Comment by mhamsa** [comment]
The migration assistant was a joke so what we ended up doing was creating new Maui projects and then started manually moving things over file by file and fixing build issues and refactoring things that wasn't completely compatible with Maui. It will probably take many weeks to get all of the issues worked out. We don't even have the solutions for our applications (2 of them) building yet. Luckily most of our business logic code was in a .NET Standard library (several hundred classes) so that was easy to migrate. But anything UI related is going to be very problematic and time consuming - the original benefits of Xamarin Forms. We've got a lot of views in our application. Very concerning for us. 

We're already telling our users NOT to upgrade to iOS 17. I'm not sure when we'll be able to support it since they're not moving Xamarin Forms forward. That's a HUGE issue in my book.

**Comment by RobSchoenaker** [comment]
Not tested, but it here is mentioned that XCode 15 beta 8 seems to work: https://github.com/xamarin/xamarin-macios/issues/18964

**Comment by IainS1986** [comment]
Can Verify the above. I've installed 16.4.0.18 on my Mac, XCode 15 (not the beta, the release) and just ran a debug build and debugger connected all fine to my iPhone.

I don't have the NewsstandKit API issue as were Xamarin native with linker turned on so it is/was being stripped anyway.

But I have Xamarin IOS, on a mac, building with Xcode 15 and debugging straight to my iOS 17 device 👍
(Note with iOS17 I do have to tap to launch the app)

(No idea if an IPA would go through app store submittion mind)

**Comment by Manish-Pradhan-FP** [comment]
I can confirm a successful debug of my Xamarin Forms app on an iOS 17 simulator using Xamarin.iOS 16.4.0.

The build _breaks_ if I keep the --weak-framework=NewsstandKit.framework/NewsstandKit argument. After removing it, the build works again.

**Comment by barantutal** [comment]
It also worked with Xamarin.iOS 16.4.0.18 and XCode 15 for me. I only had to change the linker settings to "don't link" because of an mtouch error, and I just have already published my app to get rid of the NewsstandKit crash. Hopefully, it worked on iPhones running iOS 17.

**Comment by FelipeCostaGualberto** [comment]
We are migrating an enterprise App to Flutter because of this, starting today.
Microsoft should rethink end of support policies.

For example: it is okay to kill VS for Mac, since VS Code is stable, but it is unthinkable to use MAUI in production for next ~2 years and kill Xamarin.

**Comment by pasha-o** [comment]
have you all seen this ticket - there is some promising work: 
https://github.com/xamarin/xamarin-macios/issues/18964

**Comment by LennoxP90** [comment]
SO far debug is working with the workaround, but would like to see an official support for the xcode15 tooling. It seems a lot of people's development environments have auto updated to Sonoma, and guess what xcode 14 wont run on Sonoma, so we are forced to update to xcode 15. This essentially breaks Xamarin. How can you guys support xcode14 only till May 2024 if Apple obviously dropped support for xcode 14 on their latest OS.

**Comment by giuseppenovielli** [comment]
https://devblogs.microsoft.com/dotnet/announcing-dotnet-maui-in-dotnet-8-rc-2/#bonus-2-xcode-15-and-android-api-34

**Comment by robertmaxted** [comment]
> https://devblogs.microsoft.com/dotnet/announcing-dotnet-maui-in-dotnet-8-rc-2/#bonus-2-xcode-15-and-android-api-34
> 
> YES, Xamarin.Forms apps can build with iOS 17, Android 14!

How? Using visual studio for Mac 

Xamarin.iOS
Version: 16.4.0.18 Visual Studio Community
Hash: 9d266025e
Branch: xcode14.3
Build date: 2023-09-06 19:52:27-0400


Xcode:
Version 15.0.1 (15A507)


Cannot deploy to iOS 17 Device?

Any advice greatly appreciated.

**Comment by jz-k** [comment]
> > https://devblogs.microsoft.com/dotnet/announcing-dotnet-maui-in-dotnet-8-rc-2/#bonus-2-xcode-15-and-android-api-34
> > YES, Xamarin.Forms apps can build with iOS 17, Android 14!
> 
> How? Using visual studio for Mac
> 
> Xamarin.iOS Version: 16.4.0.18 Visual Studio Community Hash: [9d26602](https://github.com/xamarin/xamarin-macios/commit/9d266025eb8b10166e0844519f0b214ede83463b) Branch: xcode14.3 Build date: 2023-09-06 19:52:27-0400
> 
> Xcode: Version 15.0.1 (15A507)
> 
> Cannot deploy to iOS 17 Device?
> 
> Any advice greatly appreciated.

Did you ever figure this out? ☹️

I am working on a Xamarin.Forms app, pulling Xamarin.iOS 16.4.0.23 into a macos-13 image on an Azure DevOps pipeline, but am still stuck with the error "Xamarin.iOS 16.4.0 does not support a deployment target of 17.0 for iOS (the maximum is 16.4)"

I am not sure what else is missing in order to target iOS 17 on build. The blog post linked above and the release notes for Xamarin.iOS 16.4.0.18+ seem to indicate that iOS 17 is supported as a build target....

**Comment by robertmaxted** [comment]
Hi @jz-k , I have since migrated to MAUI and all is working as expected.. With the deadline coming up soon for Xamarin entering end of life it is about your only option mate. Best of luck 👍

**Comment by pcervera7747** [comment]
Hi everyone, I'm not entirely clear on what the end of support on May 1, 2024 means.
Does it mean that my Xamarin.Forms application, currently running on iOS 17, will stop working?
Does it mean that my application won't work on iOS 18 or future Android versions ?

In other words, what exactly will happen if I don't migrate my Xamarin.Forms application to MAUI?

Thanks in advance!

**Comment by IainS1986** [comment]
Your app won't just stop working. 

You may also be able to maintain it for a bit, at least likely (although not guaranteed) until ios 18 and/or the next major XCode release (XCode 16?).

At that point, Xamarin.iOS will likely not be able to target XCode16, and the app stores will likely require apps built in XCode 16.

Your existing app will continue to work until Apple make some change that requires apps to be rebuilt, this may never happen (only really happened in the switch over form 32 -> 64bit).

However, if you don't switch over to .net-ios you will likely be unable to update your app as you won't be able to build an ipa that gets accepted by the app stores.

**Comment by IainS1986** [comment]
If at any point Apple or Android make some requirements that you must support in new app updates *after* May, then there is no support planned for Xamarin to allow you to do that, and porting to .net-ios or .net-android will be the only option for you.

**Comment by pcervera7747** [comment]
> If at any point Apple or Android make some requirements that you must support in new app updates _after_ May, then there is no support planned for Xamarin to allow you to do that, and porting to .net-ios or .net-android will be the only option for you.

Thank you very much, for your response @IainS1986
It was more or less my suspicion but I really appreciate you confirming it.
It seems there's no other option but to migrate sooner or later.
Best regards!

**Comment by jz-k** [comment]
> Hi @jz-k , I have since migrated to MAUI and all is working as expected.. With the deadline coming up soon for Xamarin entering end of life it is about your only option mate. Best of luck 👍

Thanks for the reply. I have work specced to transition to Maui but was hoping to get one last release on Xamarin to provide some buffer time. A bit frustrating because it seems like this should work (since xcode 15 build tools were added in Xamarin.iOS 16.4.0.18).

I must be missing something, but ultimately you're right, Xamarin's end of life is going to force a migration either way.

**Comment by angularsen** [comment]
For Azure Pipelines users, this helped me: 
https://learn.microsoft.com/en-us/answers/questions/1572351/how-to-setup-devops-pipeline-for-xamarin-forms-ios?source=docs 


Use `macOS-13` or newer, which supports iOS 17 and Android 14:
```yaml
  - stage: BuildAndroidDev
    displayName: '🔨 Android | Dev'
    pool:
      vmImage: 'macOS-latest' # Previously 'macOS-12'
```

Install Xamarin, since `macOS-12` was the last version to bundle it:
```yaml
      # Xamarin no longer included, last VM image that included it was macOS-12.
      # https://learn.microsoft.com/en-us/answers/questions/1572351/how-to-setup-devops-pipeline-for-xamarin-forms-ios?source=docs
      - task: Bash@3
        displayName: Install Xamarin
        inputs:
          targetType: 'inline'
          script: |
            brew install --cask xamarin-ios
            brew install --cask xamarin-android
```

**Comment by najak3d** [comment]
> For Azure Pipelines users, this helped me: https://learn.microsoft.com/en-us/answers/questions/1572351/how-to-setup-devops-pipeline-for-xamarin-forms-ios?source=docs
> 
> Use `macOS-13` or newer, which supports iOS 17 and Android 14:
> 
> ```yaml
> - stage: Build
>   pool:
>     vmImage: 'macOS-latest' # Previously 'macOS-12'
> ```
> 
> Install Xamarin, since `macOS-12` was the last version to bundle it:
> 
> ```yaml
>   jobs:
>     - job: Build
>       steps:
> 
>         # Xamarin no longer included, last VM image that included it was macOS-12.
>         # https://learn.microsoft.com/en-us/answers/questions/1572351/how-to-setup-devops-pipeline-for-xamarin-forms-ios?source=docs
>         - task: Bash@3
>           displayName: Install Xamarin
>           inputs:
>             targetType: 'inline'
>             script: |
>               brew install --cask xamarin-ios
>               brew install --cask xamarin-android
> ```
> 
> Then build Android/iOS as ususal:
> 
> ```yaml
>         - task: XamarinAndroid@1
>           displayName: 'Build and sign .aab file'
>           inputs:
>           # ...and more
> 
>         - task: XamariniOS@2
>           displayName: Build .ipa file
>           inputs:
>           # ...and more
> ```

Is this still an option?   We simply didn't notice the severity of all of this until just now - when we tried to publish a quick update to our iOS app and got the "Invalid Binary" return - and upon not being able to simply change the SDK from16.4 to 17 -- we then discovered that "we're screwed".

We make VERY LITTLE use of Xamarin.Forms.   Our App consists only of a SINGLE full-form SKGLView - and we render our entire UI manually via SKIA Sharp GL...  This works great, as it runs 100% the same everywhere (like a game, not real use of any native controls, other than this single Image box of type SKGLView).

So need NOTHING FROM MAUI - just need to be able to build our app against iOS SDK 17 (and XCode 15).

We've never used AZURE-anything.  We simply use Visual Studio 2022 on local PC's and Mac Minis to do our builds for Windows, Android, and iOS.   

But we're currently screwed, unable to build against SDK 17, and thus unable to release to App Store a critical patch release.  The timing of this was tragic - we unknowingly missed the cutoff by just 2 days.  The patch is a critical stability patch - but now we can't get this out to our users - until we get a workaround.

Have spent 20 hrs trying to migrate all 3 to MAUI - but this appears horrific.

Next Step will be to just try "minimal migration to MAUI" for iOS-only.

**Comment by JohnHDev** [comment]
> > For Azure Pipelines users, this helped me: https://learn.microsoft.com/en-us/answers/questions/1572351/how-to-setup-devops-pipeline-for-xamarin-forms-ios?source=docs
> > Use `macOS-13` or newer, which supports iOS 17 and Android 14:
> > ```yaml
> > - stage: Build
> >   pool:
> >     vmImage: 'macOS-latest' # Previously 'macOS-12'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Install Xamarin, since `macOS-12` was the last version to bundle it:
> > ```yaml
> >   jobs:
> >     - job: Build
> >       steps:
> > 
> >         # Xamarin no longer included, last VM image that included it was macOS-12.
> >         # https://learn.microsoft.com/en-us/answers/questions/1572351/how-to-setup-devops-pipeline-for-xamarin-forms-ios?source=docs
> >         - task: Bash@3
> >           displayName: Install Xamarin
> >           inputs:
> >             targetType: 'inline'
> >             script: |
> >               brew install --cask xamarin-ios
> >               brew install --cask xamarin-android
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Then build Android/iOS as ususal:
> > ```yaml
> >         - task: XamarinAndroid@1
> >           displayName: 'Build and sign .aab file'
> >           inputs:
> >           # ...and more
> > 
> >         - task: XamariniOS@2
> >           displayName: Build .ipa file
> >           inputs:
> >           # ...and more
> > ```
> 
> Is this still an option? We simply didn't notice the severity of all of this until just now - when we tried to publish a quick update to our iOS app and got the "Invalid Binary" return - and upon not being able to simply change the SDK from16.4 to 17 -- we then discovered that "we're screwed".
> 
> We make VERY LITTLE use of Xamarin.Forms. Our App consists only of a SINGLE full-form SKGLView - and we render our entire UI manually via SKIA Sharp GL... This works great, as it runs 100% the same everywhere (like a game, not real use of any native controls, other than this single Image box of type SKGLView).
> 
> So need NOTHING FROM MAUI - just need to be able to build our app against iOS SDK 17 (and XCode 15).
> 
> We've never used AZURE-anything. We simply use Visual Studio 2022 on local PC's and Mac Minis to do our builds for Windows, Android, and iOS.
> 
> But we're currently screwed, unable to build against SDK 17, and thus unable to release to App Store a critical patch release. The timing of this was tragic - we unknowingly missed the cutoff by just 2 days. The patch is a critical stability patch - but now we can't get this out to our users - until we get a workaround.
> 
> Have spent 20 hrs trying to migrate all 3 to MAUI - but this appears horrific.
> 
> Next Step will be to just try "minimal migration to MAUI" for iOS-only.

This is specifically if you use ADO to build, you can still make the changes locally and rebuild and deploy a Xamarin app. You might just need to install Xcode 15 and target that for the build (project options, iOS, Build, SDK Version 17.2). Xamarin Forms was updated some time back to support Xcode 15, so check the nuget versions too.
